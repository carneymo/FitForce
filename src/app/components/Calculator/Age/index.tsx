import { Box, FormControl, Grid, Input, InputAdornment } from '@mui/material';
import React, { memo, useCallback, useState } from 'react';
import debounce from 'lodash.debounce';

interface Props {
  updateAge: (age: number | string) => void;
}

export const Age = memo((props: Props) => {
  const [age, setAge] = useState('18');

  const ageCheck = event => {
    const inputAge = event.target.value.trim();
    const parsedAge = parseInt(inputAge, 10);

    if (isNaN(parsedAge)) {
      setAge('18');
      props.updateAge('18');
    } else if (parsedAge < 18) {
      setAge('18');
      props.updateAge('18');
    } else {
      setAge(inputAge);
      props.updateAge(inputAge);
    }
  };

  // eslint-disable-next-line react-hooks/exhaustive-deps
  const debouncedChangeHandler = useCallback(debounce(ageCheck, 300), []);

  const handleAgeChange = event => {
    let age = event.target.value;
    if (age.length > 2) {
      return;
    }
    // Check if the input has non-numeric characters, for example "1a" should not be allowed
    // Use regular expressions to check if the input is a number
    if (!/^\d+$/.test(age)) {
      return;
    }
    if (age.length > 0 && isNaN(parseInt(age, 10))) {
      return;
    }

    setAge(age);
    debouncedChangeHandler(event);
  };

  return (
    <Grid
      xs={4}
      item
      sx={{ marginTop: 1, textAlign: 'center' }}
      alignContent={'center'}
    >
      <Box>
        <FormControl size="small" sx={{ maxWidth: 120 }}>
          <Input
            id="standard-adornment-weight"
            endAdornment={
              <InputAdornment position="end">yrs old</InputAdornment>
            }
            value={age}
            onChange={handleAgeChange}
            onBlur={debouncedChangeHandler}
            aria-describedby="standard-weight-helper-text"
            inputProps={{
              'aria-label': 'weight',
            }}
            size="small"
          />
        </FormControl>
      </Box>
    </Grid>
  );
});
