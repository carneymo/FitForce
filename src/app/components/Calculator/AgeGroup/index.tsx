import { Grid, Box, Chip } from '@mui/material';
import * as React from 'react';
import Woman from '@mui/icons-material/Woman';
import Man from '@mui/icons-material/Man';

interface Props {
  age: number | string;
  gender: string;
}

export function AgeGroup(props: Props) {
  let age = props.age;
  let gender = props.gender;

  var ageRange = '';
  if (Number(age) < 25) {
    ageRange = '18-24';
  } else if (Number(age) < 30) {
    ageRange = '25-29';
  } else if (Number(age) < 35) {
    ageRange = '30-34';
  } else if (Number(age) < 40) {
    ageRange = '35-39';
  } else if (Number(age) < 45) {
    ageRange = '40-44';
  } else if (Number(age) < 50) {
    ageRange = '45-49';
  } else if (Number(age) < 55) {
    ageRange = '50-54';
  } else if (Number(age) < 60) {
    ageRange = '55-59';
  } else if (Number(age) >= 60) {
    ageRange = '60+';
  } else {
    ageRange = '';
  }

  let genderIcons = {
    M: <Man />,
    F: <Woman />,
  };

  return (
    <Grid
      xs={4}
      item
      sx={{ marginTop: 1, textAlign: 'center' }}
      alignContent={'center'}
    >
      <Box>
        <Chip
          icon={genderIcons[gender]}
          label={'Age Group: ' + ageRange}
          size="small"
          color={gender === 'M' ? 'primary' : 'secondary'}
        />
      </Box>
    </Grid>
  );
}
