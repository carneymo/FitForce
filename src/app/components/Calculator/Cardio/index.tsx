import {
  Card,
  Container,
  Grid,
  ToggleButton,
  ToggleButtonGroup,
} from '@mui/material';
import React, { memo } from 'react';
import DirectionsRunIcon from '@mui/icons-material/DirectionsRun';
import TransferWithinAStationIcon from '@mui/icons-material/TransferWithinAStation';
import { MileRunInput } from './MileRunInput';
import { HamrInput } from './HamrInput';

interface Props {
  age: number | string;
  gender: string;
}

export const Cardio = memo((props: Props) => {
  let age = props.age;
  let gender = props.gender;

  const [cardioType, setCardioType] = React.useState('run');

  const handleChange = (
    event: React.MouseEvent<HTMLElement>,
    newCardioType: string,
  ) => {
    setCardioType(newCardioType);
  };

  return (
    <Container
      sx={{
        paddingTop: 3,
        height: '100vh',
      }}
    >
      <Card elevation={3} sx={{ padding: 2, fontSize: '10pt' }}>
        <Grid container spacing={2}>
          <Grid item xs={2} sm={2}>
            <ToggleButtonGroup
              orientation="vertical"
              value={cardioType}
              exclusive
              onChange={handleChange}
              sx={{ textAlign: 'left' }}
            >
              <ToggleButton value="run" aria-label="run">
                <DirectionsRunIcon />
              </ToggleButton>
              <ToggleButton value="hamr" aria-label="hamr">
                <TransferWithinAStationIcon />
              </ToggleButton>
            </ToggleButtonGroup>
          </Grid>
          <Grid item xs={10} sm={10}>
            {cardioType === 'run' ? (
              <MileRunInput age={age} gender={gender} />
            ) : (
              <HamrInput age={age} gender={gender} />
            )}
          </Grid>
        </Grid>
      </Card>
    </Container>
  );
});
