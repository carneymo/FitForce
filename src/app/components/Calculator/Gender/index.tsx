import { Grid, ToggleButton, ToggleButtonGroup } from '@mui/material';
import React, { memo, useState } from 'react';

interface Props {
  updateGender: (gender: string) => void;
}

export const Gender = memo((props: Props) => {
  const [gender, setGender] = useState<string>('M'); // State to store the selected gender

  const handleGenderChange = (
    event: React.MouseEvent<HTMLElement>,
    newGender: string | null,
  ) => {
    if (newGender) {
      setGender(newGender); // Update the selected gender in the state
      props.updateGender(newGender); // Call the updateGender function from props with the new gender value
    }
  };

  return (
    <Grid
      xs={4}
      item
      display={'flex'}
      alignItems={'center'}
      justifyContent={'center'}
    >
      <ToggleButtonGroup
        color="primary"
        exclusive
        value={gender}
        onChange={handleGenderChange}
        aria-label="Gender"
        size="small"
      >
        <ToggleButton value="M">Male</ToggleButton>
        <ToggleButton value="F">Female</ToggleButton>
      </ToggleButtonGroup>
    </Grid>
  );
});
