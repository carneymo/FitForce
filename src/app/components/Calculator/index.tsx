import React from 'react';

import { Cardio } from './Cardio';
import { Gender } from './Gender';
import { Age } from './Age';
import { Card, Container, Grid, Paper, Typography } from '@mui/material';
import { AgeGroup } from './AgeGroup';
import CalculateIcon from '@mui/icons-material/Calculate';

interface Props {}

interface State {
  age: number | string;
  gender: string;
}

class Calculator extends React.Component<{}, State> {
  constructor(props: Props) {
    super(props);
    this.state = {
      age: 18,
      gender: 'M',
    };
    this.updateGender = this.updateGender.bind(this);
    this.updateAge = this.updateAge.bind(this);
  }

  componentDidMount(): void {
    console.log('Calculator Mounted');
  }

  updateGender(genderType: string) {
    this.setState({ gender: genderType });
    console.log('Updated Gender: ' + genderType);
  }

  updateAge(age: string | number) {
    this.setState({ age: age });
    console.log('Updated Age: ' + age);
  }

  render() {
    return (
      <Paper>
        <Container
          sx={{
            paddingTop: 3,
            height: '100vh',
          }}
        >
          <Card elevation={3} sx={{ padding: 2, fontSize: '10pt' }}>
            <Grid container spacing={2}>
              <Grid item xs={12}>
                <div
                  style={{
                    display: 'flex',
                    alignItems: 'center',
                    flexWrap: 'wrap',
                  }}
                >
                  <CalculateIcon
                    color="primary"
                    style={{ marginRight: '15px' }}
                  />
                  <strong>PFT Calculator</strong>
                </div>
              </Grid>
              <Grid item xs={12}>
                <Typography variant="body2">
                  This calculator is used to determine your PFT score. The
                  formulas used are based on the Air Force Instruction 36-2905,
                  Fitness Program.
                </Typography>
              </Grid>
            </Grid>
          </Card>

          <Grid container rowSpacing={1} marginTop={2}>
            <Gender updateGender={this.updateGender} />
            <Age updateAge={this.updateAge} />
            <AgeGroup age={this.state.age} gender={this.state.gender} />
          </Grid>
          <Grid container spacing={2} marginTop={3} marginBottom={3}>
            <Cardio age={this.state.age} gender={this.state.gender} />
          </Grid>
        </Container>
      </Paper>
    );
  }
}

export default Calculator;
