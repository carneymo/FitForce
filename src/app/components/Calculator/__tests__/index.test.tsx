import * as React from 'react';
import { render } from '@testing-library/react';
import Calculator from '..';

describe('<Calculator  />', () => {
  it('should match snapshot', () => {
    const loadingIndicator = render(<Calculator />);
  });
});
