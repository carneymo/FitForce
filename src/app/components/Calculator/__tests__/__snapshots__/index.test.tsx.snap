// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Calculator  /> should match snapshot 1`] = `
<div
  class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1ps6pg7-MuiPaper-root"
>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-1o6ysxg-MuiContainer-root"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 MuiCard-root css-sc5siq-MuiPaper-root-MuiCard-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
        >
          <div
            style="display: flex; align-items: center; flex-wrap: wrap;"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-colorSecondary MuiSvgIcon-fontSizeMedium css-kw0y8f-MuiSvgIcon-root"
              data-testid="CalculateIcon"
              focusable="false"
              style="margin-right: 15px;"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-5.97 4.06L14.09 6l1.41 1.41L16.91 6l1.06 1.06-1.41 1.41 1.41 1.41-1.06 1.06-1.41-1.4-1.41 1.41-1.06-1.06 1.41-1.41-1.41-1.42zm-6.78.66h5v1.5h-5v-1.5zM11.5 16h-2v2H8v-2H6v-1.5h2v-2h1.5v2h2V16zm6.5 1.25h-5v-1.5h5v1.5zm0-2.5h-5v-1.5h5v1.5z"
              />
            </svg>
            <strong>
              PFT Calculator
            </strong>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          >
            This calculator is used to determine your PFT score. The formulas used are based on the Air Force Instruction 36-2905, Fitness Program.
          </p>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container css-1k1z3r7-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-1xyxrxl-MuiGrid-root"
      >
        <div
          aria-label="Gender"
          class="MuiToggleButtonGroup-root css-1gjgmky-MuiToggleButtonGroup-root"
          role="group"
        >
          <button
            aria-pressed="true"
            class="MuiButtonBase-root MuiToggleButton-root Mui-selected MuiToggleButton-sizeSmall MuiToggleButton-primary MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal css-1dlyelz-MuiButtonBase-root-MuiToggleButton-root"
            tabindex="0"
            type="button"
            value="M"
          >
            Male
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            aria-pressed="false"
            class="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeSmall MuiToggleButton-primary MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal css-1dlyelz-MuiButtonBase-root-MuiToggleButton-root"
            tabindex="0"
            type="button"
            value="F"
          >
            Female
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-tia25t-MuiGrid-root"
      >
        <div
          class="MuiBox-root css-0"
        >
          <div
            class="MuiFormControl-root css-1fyavwt-MuiFormControl-root"
          >
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall MuiInputBase-adornedEnd css-v4u5dn-MuiInputBase-root-MuiInput-root"
            >
              <input
                aria-describedby="standard-weight-helper-text"
                aria-invalid="false"
                aria-label="weight"
                class="MuiInputBase-input MuiInput-input MuiInputBase-inputSizeSmall MuiInputBase-inputAdornedEnd css-nz481w-MuiInputBase-input-MuiInput-input"
                id="standard-adornment-weight"
                type="text"
                value="18"
              />
              <div
                class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeSmall css-1laqsz7-MuiInputAdornment-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1pnmrwp-MuiTypography-root"
                >
                  yrs old
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-tia25t-MuiGrid-root"
      >
        <div
          class="MuiBox-root css-0"
        >
          <div
            class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorDefault MuiChip-filledDefault css-1fj5pxn-MuiChip-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiChip-icon MuiChip-iconSmall MuiChip-iconColorDefault css-i4bv87-MuiSvgIcon-root"
              data-testid="ManIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M14 7h-4c-1.1 0-2 .9-2 2v6h2v7h4v-7h2V9c0-1.1-.9-2-2-2z"
              />
              <circle
                cx="12"
                cy="4"
                r="2"
              />
            </svg>
            <span
              class="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
            >
              Age Group: 18-24
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-3syr29-MuiGrid-root"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg css-1o6ysxg-MuiContainer-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 MuiCard-root css-sc5siq-MuiPaper-root-MuiCard-root"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-2 css-1o7apob-MuiGrid-root"
            >
              <div
                class="MuiToggleButtonGroup-root MuiToggleButtonGroup-vertical css-1g71zaa-MuiToggleButtonGroup-root"
                role="group"
              >
                <button
                  aria-label="run"
                  aria-pressed="true"
                  class="MuiButtonBase-root MuiToggleButton-root Mui-selected MuiToggleButton-sizeMedium MuiToggleButton-standard MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedVertical css-ueukts-MuiButtonBase-root-MuiToggleButton-root"
                  tabindex="0"
                  type="button"
                  value="run"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="DirectionsRunIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M13.49 5.48c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm-3.6 13.9 1-4.4 2.1 2v6h2v-7.5l-2.1-2 .6-3c1.3 1.5 3.3 2.5 5.5 2.5v-2c-1.9 0-3.5-1-4.3-2.4l-1-1.6c-.4-.6-1-1-1.7-1-.3 0-.5.1-.8.1l-5.2 2.2v4.7h2v-3.4l1.8-.7-1.6 8.1-4.9-1-.4 2 7 1.4z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
                <button
                  aria-label="hamr"
                  aria-pressed="false"
                  class="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeMedium MuiToggleButton-standard MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedVertical css-ueukts-MuiButtonBase-root-MuiToggleButton-root"
                  tabindex="0"
                  type="button"
                  value="hamr"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="TransferWithinAStationIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M16.49 15.5v-1.75L14 16.25l2.49 2.5V17H22v-1.5zm3.02 4.25H14v1.5h5.51V23L22 20.5 19.51 18zM9.5 5.5c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zM5.75 8.9 3 23h2.1l1.75-8L9 17v6h2v-7.55L8.95 13.4l.6-3C10.85 12 12.8 13 15 13v-2c-1.85 0-3.45-1-4.35-2.45l-.95-1.6C9.35 6.35 8.7 6 8 6c-.25 0-.5.05-.75.15L2 8.3V13h2V9.65l1.75-.75"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 MuiGrid-grid-sm-10 css-17p3wh3-MuiGrid-root"
            >
              <div
                class="MuiBox-root css-fgvrm7"
              >
                <strong>
                  1.5 Mile Run
                </strong>
                <div
                  class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
                >
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 css-19gy2ei-MuiGrid-root"
                  >
                    <div
                      class="MuiFormControl-root MuiFormControl-marginDense MuiTextField-root css-13e98gt-MuiFormControl-root-MuiTextField-root"
                    >
                      <label
                        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
                        data-shrink="true"
                        for="milerun-minutes"
                        id="milerun-minutes-label"
                      >
                        mm
                      </label>
                      <div
                        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
                      >
                        <input
                          aria-invalid="false"
                          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                          id="milerun-minutes"
                          type="text"
                          value="00"
                        />
                        <fieldset
                          aria-hidden="true"
                          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                        >
                          <legend
                            class="css-14lo706"
                          >
                            <span>
                              mm
                            </span>
                          </legend>
                        </fieldset>
                      </div>
                    </div>
                    <div
                      class="MuiFormControl-root MuiFormControl-marginDense MuiTextField-root css-xnm3t8-MuiFormControl-root-MuiTextField-root"
                    >
                      <label
                        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
                        data-shrink="true"
                        for="milerun-seconds"
                        id="milerun-seconds-label"
                      >
                        ss
                      </label>
                      <div
                        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
                      >
                        <input
                          aria-invalid="false"
                          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                          id="milerun-seconds"
                          type="text"
                          value="00"
                        />
                        <fieldset
                          aria-hidden="true"
                          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                        >
                          <legend
                            class="css-14lo706"
                          >
                            <span>
                              ss
                            </span>
                          </legend>
                        </fieldset>
                      </div>
                    </div>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 css-fjbua9-MuiGrid-root"
                  >
                    <p
                      class="MuiTypography-root MuiTypography-body1 css-e93pxh-MuiTypography-root"
                    >
                      5.0
                    </p>
                     
                    <p
                      class="MuiTypography-root MuiTypography-body1 css-vappet-MuiTypography-root"
                    >
                      / 20
                    </p>
                    <div
                      class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorError MuiChip-filledError css-lwe8th-MuiChip-root"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiChip-icon MuiChip-iconSmall MuiChip-iconColorError css-i4bv87-MuiSvgIcon-root"
                        data-testid="WarningAmberIcon"
                        focusable="false"
                        style="color: white;"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 5.99 19.53 19H4.47L12 5.99M12 2 1 21h22L12 2z"
                        />
                        <path
                          d="M13 16h-2v2h2zm0-6h-2v5h2z"
                        />
                      </svg>
                      <span
                        class="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
                      >
                        Unsatisfactory
                      </span>
                    </div>
                    <div
                      class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorPrimary MuiChip-filledPrimary css-174hymm-MuiChip-root"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiChip-icon MuiChip-iconSmall MuiChip-iconColorPrimary css-i4bv87-MuiSvgIcon-root"
                        data-testid="RecommendIcon"
                        focusable="false"
                        style="color: white;"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 2a10 10 0 1 0 10 10A10 10 0 0 0 12 2zm6 9.8a.9.9 0 0 1-.1.5l-2.1 4.9a1.34 1.34 0 0 1-1.3.8H9a2 2 0 0 1-2-2v-5a1.28 1.28 0 0 1 .4-1L12 5l.69.69a1.08 1.08 0 0 1 .3.7v.2L12.41 10H17a1 1 0 0 1 1 1z"
                        />
                      </svg>
                      <span
                        class="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
                      >
                        Satisfactory
                      </span>
                    </div>
                    <div
                      class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorSuccess MuiChip-filledSuccess css-li422f-MuiChip-root"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiChip-icon MuiChip-iconSmall MuiChip-iconColorSuccess css-i4bv87-MuiSvgIcon-root"
                        data-testid="StarIcon"
                        focusable="false"
                        style="color: gold;"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                        />
                      </svg>
                      <span
                        class="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
                      >
                        Excellent
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
